android/WALT/app/src/main/java/org/chromium/latency/walt/ClockManager.java

```diff
@@ -349,6 +352,19 @@ public String readAll() throws IOException {
         return s;
     }
 
+    public void checkVersion() throws IOException {
+        if (!isConnected()) throw new IOException("Not connected to WALT");
+        if (!isListenerStopped()) throw new IOException("Listener is running");
+
+        String s = command(CMD_VERSION);
+        mLogger.log("WALT reported protocol version " + s);
+        if (!PROTOCOL_VERSION.equals(s)) {
+            throw new IOException("The firmware version reported by WALT is not supported.\n"
```

Comment:
We should print both the reported and the expected protocol version, it can sometimes happen that the reported version will be above expected, say if the phone was laying around for a while and the app was not updated.

We should also avoid using "firmware version", it should be strictly "protocol version",  for firmware version it feels like it should be incremented with every new release and no two releases should have the same version number, while we only intend to increment this one when there is breakage in protocol or some good reason to force everyone to upgrade.